src/node_crypto.h

```diff
@@ -142,13 +142,6 @@ class SecureContext : public BaseObject {
   void FreeCTXMem() {
     if (ctx_) {
       env()->isolate()->AdjustAmountOfExternalAllocatedMemory(-kExternalSize);
-      if (ctx_->cert_store == root_cert_store) {
-        // SSL_CTX_free() will attempt to free the cert_store as well.
-        // Since we want our root_cert_store to stay around forever
-        // we just clear the field. Hopefully OpenSSL will not modify this
-        // struct in future versions.
-        ctx_->cert_store = nullptr;
```

Comment:
On 08/30/2016 06:44 PM, Fedor Indutny wrote:
> In src/node_crypto.h
> <https://github.com/nodejs/node/pull/8334#discussion_r76833196>:
>
>> @@ -142,13 +142,6 @@ class SecureContext : public BaseObject {
>>    void FreeCTXMem() {
>>      if (ctx_) {
>>        env()->isolate()->AdjustAmountOfExternalAllocatedMemory(-kExternalSize);
>> -      if (ctx_->cert_store == root_cert_store) {
>> -        // SSL_CTX_free() will attempt to free the cert_store as well.
>> -        // Since we want our root_cert_store to stay around forever
>> -        // we just clear the field. Hopefully OpenSSL will not modify this
>> -        // struct in future versions.
>> -        ctx_->cert_store = nullptr;
>
> Why isn't this necessary anymore?

See the commit message that touches this. Basically, using references 
count makes this no longer necessary. The OpenSSL free function checks 
that references > 1 and does not free the cert store. The free function 
does decrement reference hence the reason why it is needed to increment 
it every time it is assigned to a SSL_CTX

- Adam